keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -396,6 +400,37 @@ type PodSpec struct {
 }
 ```
 
+#### PodStatus API changes
+
+Extend `PodStatus` to include pod-level analog of the container status resource
+fields. Pod-level resource information in `PodStatus` is essential for pod-level [In-Place Pod
+Update]
+(https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/1287-in-place-update-pod-resources/README.md#api-changes)
+as it provides a way to track, report and use the actual resource allocation for the
+pod, both before and after a resize operation.
+
+```
+type PodStatus struct {
+...
+  // Resources represents the compute resource requests and limits that have been
+  // applied at the pod level. If pod-level resources are not explicitly specified,
+  // then these will be the aggregate resources computed from containers. If limits are
+  // not defined for all containers (and not defined at the pod level) then no aggregate
+  // pod-level limits will be applied.
+  // +featureGate=InPlacePodVerticalScaling
+  // +optional
+  Resources *ResourceRequirements
+
+  // AllocatedResources is the total requests allocated for this pod by the node.
+  // Kubelet sets this to the accepted requests when a pod (or resize) is admitted.
+  // If pod-level requests are not set, this will be the total requests aggregated
+  // across containers in the pod.
+  // +featureGate=InPlacePodVerticalScaling
+  // +optional
+  AllocatedResources ResourceList
```

Comment:
pointer for new optional field?

```suggestion
  AllocatedResources *ResourceList
```

Reply:
Ah, I didn't chase the type. As long as it doesn't serialize anything in JSON or proto when empty, that's fine